### PR TITLE
Travis CI: Try npm 5.7's `npm ci` to install dependencies 

### PR DESCRIPTION
**Pull request by ntwb**
## Description

npm 5.7 includes a new `ci` _install_ command, this `ci` command is designed to be used by _continous integration_ services such as Travis CI and Circle CI.

> http://blog.npmjs.org/post/171139955345/v570
> The new `npm ci` command installs from your lock-file ONLY. If your `package.json` and your lock-file are out of sync then it will report an error.
> 
> It works by throwing away your `node_modules` and recreating it from scratch.
> 
> Beyond guaranteeing you that you’ll only get what is in your lock-file it’s also much faster (2x-10x!) than `npm install` when you don’t start with a `node_modules`.
> 
> As you may take from the name, we expect it to be a big boon to continuous integration environments. We also expect that folks who do production deploys from git tags will see major gains.

As we use a `package-lock.json` file for Gutenberg and we don't cache `node_modules` on Travis CIs cache we should have a shot at this 😄 

One downside is we already have a npm script command named `ci` ([src](https://github.com/WordPress/gutenberg/blob/master/package.json#L151)) which makes the following that is implement in this PR a little confusing:
```yml
        - npm ci || exit 1
        - npm run ci || exit 1	        - npm run ci || exit 1
```

If this PR is merged maybe renaming that existing `ci` npm script would be worthwhile to avoid that confusion

## How Has This Been Tested?

Travis CI performance results will be posted in follow up comments

## Types of changes

CI testing performance

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code has proper inline documentation.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ cache:
 
 before_install:
   - nvm install && nvm use
-  - npm install npm@latest -g
+  - npm install npm@5.7.1 -g
```
Comment (ntwb) [inline .travis.yml]: Via npm: _"npm@latest is going to be 5.6.0 for the next week or two. 5.7.1 is released under npm@next -- we stagger all our releases this way, to give early adopters time to report major bad stuff"_

**Comment by ntwb** [comment]
This PR Build 16553: https://travis-ci.org/WordPress/gutenberg/builds/345136618
• Job #16553.1 - 55.94s
• Job #16553.3 - 74.85s
(The above tests also adds an extra ~12 seconds to install npm 5.7.1)

Build 16551: https://travis-ci.org/WordPress/gutenberg/builds/345131531
• Job #16551.1 - 50.78s
• Job #16551.3 - 53.91s

The results, `.1` jobs an extra ~17 seconds, `.3` jobs an extra ~33 seconds 😭 

Closing this PR, maybe later...

**Comment by zkat** [comment]
Hey! So I looked into this after adding some instrumentation to `npm ci`. As it turns out, I think changing npm installers is unlikely to have a massive effect for you: a hefty *_40s_* or more is spent just downloading Cypress -- and I wouldn't be surprised if the multi-second variance you're seeing in _that_ is due to network hiccups on their own distribution downloads.

You might wanna try enabling the npm cache by caching `$HOME/.npm` in your `.travis.yml`, which might in general shave off a few seconds (both for `npm install` and `npm ci`), but that bigass Cypress download is gonna loom over everything else :\

**Comment by ntwb** [comment]
Thanks @zkat, and yes, not being able to cache that Cypress download is a pain

Hopefully, https://github.com/cypress-io/cypress/issues/1300 gets a PR soon and it'll no longer be an issue 🤞

**Pull request closed without merging**